packages/scanner/src/cli/scan/watchScan.ts

```diff
@@ -44,16 +46,56 @@ function isAncestorPath(ancestor: string, descendant: string): boolean {
   return !path.relative(ancestor, descendant).startsWith('..');
 }
 
+export type BatchedTelemetry = {
+  telemetryName: string;
+  scanResults: ScanResults;
+  msElapsed: number;
+  msInserted: number;
+};
+
 export class Watcher {
   config?: TimestampedConfiguration;
   appmapWatcher?: chokidar.FSWatcher;
   appmapPoller?: chokidar.FSWatcher;
   configWatcher?: chokidar.FSWatcher;
+  batchedTelemetry: BatchedTelemetry[] = [];
+  sendPeriodMilliseconds = 5000;
+  interval: NodeJS.Timer;
+
+  constructor(private options: WatchScanOptions) {
+    this.interval = setInterval(() => {
+      this.sendScanResultsTelemetryInABatch();
+    }, this.sendPeriodMilliseconds);
+    // must unref because there's no condition under which the timer
+    // will stop; else the process doesn't exit
+    this.interval.unref();
+  }
 
-  constructor(private options: WatchScanOptions) {}
+  public sendScanResultsTelemetryInABatch() {
+    const dateNow = Date.now();
+    if (
+      this.batchedTelemetry &&
+      this.batchedTelemetry.length > 0 &&
+      dateNow - this.batchedTelemetry[this.batchedTelemetry.length - 1].msInserted >
```

Comment:
What is the purpose of this clause? Why don't you just send all the telemetry that's in the Array?

Reply:
This test seems wrong to me. If the last `batchedTelemetry` event is inserted at `dateNow` then `dateNow - dateNow > this.sendPeriodMilliseconds` will be false.

Reply:
1) The purpose is to send data after a period of inactivity.  Not to send data at a specific period.  If data are being collected for 10 minutes, and then there's inactivity for 5 seconds, data will be sent at 10:05.

I [believe](https://github.com/getappmap/appmap-js/pull/847#discussion_r1029606874) it's better to send data every 5 seconds instead of after a period of inactivity.

2) 
```
# this
dateNow - dateNew > this.sendPeriodMilliseconds

# reduces to
0 > 5000

# reduces to
false
```